### DISCUSSION TITLE
Compress data for upload

### DISCUSSION THREAD

**Discussion by dlintw** [asciinema/discussions#13] [Ideas]
My upload bandwidth is limited.
Could the upload command send by compressed data instead of json text format?

**Comment by ku1ik** [comment]
How big is the file you try to upload?

**Comment by dlintw** [comment]
Well, 5MB on 64k upload bandwidth..

**Comment by Jellyfrog** [comment]
Gziping the output would yield a >90% compression ratio.
Maybe its a good idea to always compress the files, even for local saved files?

**Comment by ku1ik** [comment]
FYI asciinema/asciinema#146 was reverted as it wasn't working.

**Comment by bkil** [comment]
Related, but not limited to upload transfer encoding (and with potential lossiness): asciinema/asciinema#515
